src/xcompact.f90

```diff
@@ -22,9 +22,9 @@ program xcompact
 
   class(base_backend_t), pointer :: backend
   class(allocator_t), pointer :: allocator
-  type(mesh_t) :: mesh
   type(allocator_t), pointer :: host_allocator
   type(solver_t) :: solver
+  type(mesh_t), target :: mesh
```

Comment:
Should this be `class` based on `src/allocator.f90`?

Reply:
`type` was correct, allocator has been fixed.